src/platform/w3dengine/client/w3dwaypointbuffer.cpp

```diff
@@ -0,0 +1,330 @@
+                                    Coord3D pos = *obj->Get_Position();
+                                    pos.Sub(obj2->Get_Position());
+
+                                    if (obj->Get_Vision_Range() >= pos.Length()) {
+                                        const AIUpdateInterface *update = obj->Get_AI_Update_Interface();
+                                        int size = update != nullptr ? update->Friend_Get_Waypoint_Goal_Path_Size() : 0;
+                                        int index = update != nullptr ? update->Get_Current_Goal_Path_Index() : 0;
+
+                                        if (update != nullptr) {
+                                            bool has_points = false;
+
+                                            const Coord3D *pos2 = obj2->Get_Position();
+                                            points[count] = Vector3(pos2->x, pos2->y, pos2->z);
+                                            count++;
+
+                                            if (index >= 0 && index < size) {
+                                                for (int i = index; i < size; i++) {
+                                                    const Coord3D *goal = update->Get_Goal_Path_Position(i);
+
+                                                    if (goal != nullptr) {
+                                                        if (count < 513) {
+                                                            points[count] = Vector3(goal->x, goal->y, goal->z);
+                                                            count++;
+                                                        }
+
+                                                        m_renderObj->Set_Position(Vector3(goal->x, goal->y, goal->z));
+                                                        W3D::Render(*m_renderObj, r);
+                                                        has_points = true;
+                                                    }
+                                                }
+                                            } else {
+                                                const Coord3D *goal = update->Get_Goal_Position();
+
+                                                if (goal->Length() > 1.0f) {
+                                                    points[count] = Vector3(goal->x, goal->y, goal->z);
+                                                    count++;
+                                                    m_renderObj->Set_Position(Vector3(goal->x, goal->y, goal->z));
+                                                    W3D::Render(*m_renderObj, r);
+                                                    has_points = true;
+                                                }
+                                            }
+
+                                            if (has_points) {
+                                                m_segLine->Set_Color(Vector3(0.94999999f, 0.5f, 0.0f));
```

Comment:
this should be 0.95f, which will be represented as 0.9499999f, due to floating point arithmetics

Reply:
See https://www.h-schmidt.net/FloatConverter/IEEE754.html for reference :)

Reply:
fixed.